sdk/identity/identity/src/credentials/authorityValidationOptions.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Provides options to configure how the Identity library
+ * does authority validation during authentication requests
+ * to Azure Active Directory.
+ */
+export interface AuthorityValidationOptions {
+  instanceDiscovery?: boolean;
+}
+
+// Avoid situation where instanceDiscovery is undefined and the condition for false is satisfied
+export function setDefaultInstanceDisovery(instanceDiscovery?: boolean): boolean {
```

Comment:
This name sounds weird, because it's a boolean function and yet it's called `set` which implies some sort of mutation.  Would something like `shouldSetDefaultDiscovery` work better?

Reply:
Api rename to `disableInstanceDiscovery` removes the need for this helper function